### ISSUE TITLE
Support for multiple, flexible goals.

### ISSUE THREAD

**Issue by kbumsik** [kbumsik/WolfieMouse#32]
This program only supports a single static goal. However, there are 4 goals in the competition.

On top of that, when you look for some Micromouse video, you can see patterns of the micromouse:

1. Search for cell while trying to reach to the goal.
2. After getting to the goal, it look for unsearched cells and then go back to the starting point.
3. Then drive to the goal directly as fast as the mouse can.

This implies we need to introduce `destination` variable that can be changed during running. 
To summarize:

- [x] Add support for multiple goals. The algorithm implementation should also be fixed accordingly.
- [x] Add `Destinations` property in `MouseController` class. Also add getter and setter methods for `Destination`. `MouseController` class should not use `Goals` directly, and make use of `Destinations` instead. The program should support multiple `Destinations` as well.

Ask me about designing this anytime. I will work for hardware driver for a week.
